### PR TITLE
chore: Update removal dates to either v1.35.0 or v1.40.0

### PR DESCRIPTION
**Pull request by powersj**
## Summary
Updates all deprecated removal messages to set them for June 2025 as a part of v1.35.0 or September 2026 with v1.40.0.

## Checklist

- [x] No AI generated code was used in this PR

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -552,7 +552,7 @@ func (c *Config) LoadConfigData(data []byte) error {
 	if c.Agent.SnmpTranslator == "netsnmp" {
 		PrintOptionValueDeprecationNotice("agent", "snmp_translator", "netsnmp", telegraf.DeprecationInfo{
 			Since:     "1.25.0",
-			RemovalIn: "2.0.0",
+			RemovalIn: "1.40.0",
```
Comment (DStrand1) [inline config/config.go:555]: Should this be 1.35.0?

File: plugins/inputs/elasticsearch/elasticsearch.go [hunk 2]
```diff
@@ -98,7 +98,7 @@ type indexStat struct {
 type Elasticsearch struct {
 	Local                      bool            `toml:"local"`
 	Servers                    []string        `toml:"servers"`
-	HTTPTimeout                config.Duration `toml:"http_timeout" deprecated:"1.29.0;use 'timeout' instead"`
+	HTTPTimeout                config.Duration `toml:"http_timeout" deprecated:"1.29.0;1.35.0;use 'timeout' instead"`
```
Comment (DStrand1) [inline plugins/inputs/elasticsearch/elasticsearch.go:101]: Should this be 1.40.0?

File: plugins/inputs/gnmi/gnmi.go [hunk 3]
```diff
@@ -58,9 +58,9 @@ type GNMI struct {
 	CanonicalFieldNames  bool              `toml:"canonical_field_names"`
 	TrimFieldNames       bool              `toml:"trim_field_names"`
 	PrefixTagKeyWithPath bool              `toml:"prefix_tag_key_with_path"`
-	GuessPathTag         bool              `toml:"guess_path_tag" deprecated:"1.30.0;use 'path_guessing_strategy' instead"`
+	GuessPathTag         bool              `toml:"guess_path_tag" deprecated:"1.30.0;1.35.0;use 'path_guessing_strategy' instead"`
```
Comment (DStrand1) [inline plugins/inputs/gnmi/gnmi.go:61]: Should this be 1.40.0?

File: plugins/parsers/binary/parser.go [hunk 4]
```diff
@@ -16,10 +16,10 @@ import (
 
 type Parser struct {
 	AllowNoMatch bool            `toml:"allow_no_match"`
-	Endianess    string          `toml:"endianess" deprecated:"1.27.4;use 'endianness' instead"`
+	Endianess    string          `toml:"endianess" deprecated:"1.27.4;1.35.0;use 'endianness' instead"`
 	Endianness   string          `toml:"endianness"`
 	Configs      []Config        `toml:"binary"`
-	HexEncoding  bool            `toml:"hex_encoding" deprecated:"1.30.0;use 'binary_encoding' instead"`
+	HexEncoding  bool            `toml:"hex_encoding" deprecated:"1.30.0;1.35.0;use 'binary_encoding' instead"`
```
Comment (DStrand1) [inline plugins/parsers/binary/parser.go:22]: Should this be 1.40.0?

**Review by DStrand1 — COMMENTED** [review]
Looks great! Just a couple questions around a few of the versions that are pretty close to eachother

File: config/config.go (hunk 1 above)
Comment (powersj) [reply config/config.go:555]: no - I meant to talk about this one in pairs, but no way we should change this next year.

File: plugins/inputs/elasticsearch/elasticsearch.go (hunk 2 above)
Comment (powersj) [inline plugins/inputs/elasticsearch/elasticsearch.go:101]:
```suggestion
	HTTPTimeout                config.Duration `toml:"http_timeout" deprecated:"1.29.0;1.40.0;use 'timeout' instead"`
```

File: plugins/inputs/gnmi/gnmi.go (hunk 3 above)
Comment (powersj) [inline plugins/inputs/gnmi/gnmi.go:61]:
```suggestion
	GuessPathTag         bool              `toml:"guess_path_tag" deprecated:"1.30.0;1.40.0;use 'path_guessing_strategy' instead"`
```

File: plugins/parsers/binary/parser.go (hunk 4 above)
Comment (powersj) [inline plugins/parsers/binary/parser.go:22]:
```suggestion
	HexEncoding  bool            `toml:"hex_encoding" deprecated:"1.30.0;1.40.0;use 'binary_encoding' instead"`
```

**Comment by powersj** [comment]
> LGTM. Thanks @powersj! We should certainly provide migrations for those options etc?!?

Yeah let's track that

**Review by DStrand1 — APPROVED** [review]

**Pull request merged**